src/auro-input.js

```diff
@@ -56,7 +56,7 @@ export default class AuroInput extends BaseInput {
         id="${this.id}"
         name="${ifDefined(this.name)}"
         type="${this.type === 'password' && this.showPassword ? 'text' : this.getInputType(this.type)}"
-        pattern="${ifDefined(this.type === 'credit-card' && !this.noValidate && this.maxLength ? `.{${this.maxLength},${this.maxLength}}` : undefined)}"
+        pattern="${ifDefined(this.pattern ? this.creditCardPattern() : undefined)}"
```

Comment:
Can we rename `this.creditCardPattern`? From the name, it's not clear that it falls back to a user defined pattern. Maybe rename to `getPattern`.

I think the whole ternary can be simplified, since `this.creditCardPattern` will return undefined if no pattern is defined.

```suggestion
        pattern="${ifDefined(this.getPattern())}"
```